src/Raven.Server/Rachis/Leader.cs

```diff
@@ -899,19 +929,26 @@ public void Dispose()
 
                         foreach (var entry in _entries)
                         {
-                            if (entry.Value.OnNotify != null)
+                            if (entry.Key <= _lastCommit)
                             {
-                                try
+                                if (entry.Value.OnNotify != null)
                                 {
-                                    entry.Value.OnNotify(entry.Value.TaskCompletionSource);
+                                    try
+                                    {
+                                        entry.Value.OnNotify(entry.Value.TaskCompletionSource);
+                                    }
+                                    catch (Exception e)
+                                    {
+                                        entry.Value.TaskCompletionSource.TrySetException(e);
+                                        continue;
```

Comment:
I think this `continue` should be at the end of the `if (entry.Key <= _lastCommit)` scope

Reply:
Done